nexus/types/src/external_api/views.rs

```diff
@@ -1571,6 +1572,53 @@ pub struct UpdatesTrustRoot {
     pub root_role: TufSignedRootRole,
 }
 
+/// Metadata about a TUF repository
+#[derive(Clone, Debug, PartialEq, Eq, Deserialize, Serialize, JsonSchema)]
+pub struct TufRepo {
+    /// The hash of the repository.
+    ///
+    /// This is a slight abuse of `ArtifactHash`, since that's the hash of
+    /// individual artifacts within the repository. However, we use it here for
+    /// convenience.
+    pub hash: ArtifactHash,
+
+    /// The version of the targets role
+    pub targets_role_version: u64,
+
+    /// The time until which the repo is valid
+    pub valid_until: DateTime<Utc>,
+
+    /// The system version in artifacts.json
+    pub system_version: Version,
+
+    /// The file name of the repository
+    ///
+    /// This is purely used for debugging and may not always be correct (e.g.,
+    /// with wicket, we read the file contents from stdin so we don't know the
+    /// correct file name).
+    pub file_name: String,
+}
```

Comment:
These are just the fields that were already on `TufRepoMeta`, but I haven't really thought about whether they're all necessary. I have no idea what `targets_role_version` is, so there's at least one candidate for removal. I also will probably want to explain each field better.

Reply:
Yeah, the public-facing docs need work here.  If we're going to do nothing else, I'd leave them undocumented and change these to regular comments.  It's just confusing to talk about artifacts.json and our "abuse of `ArtifactHash`" in public-facing docs.